dynawo/sources/Models/CPP/Events/ModelVoltageSetPointChange/test/TestVoltageSetPointChange.cpp

```diff
@@ -0,0 +1,216 @@
+  for (size_t i = 0; i < modelVoltageSetPointChange->sizeZ(); ++i)
+    zConnected[i] = true;
+  modelVoltageSetPointChange->setBufferZ(&z[0], zConnected, 0);
+  modelVoltageSetPointChange->init(0);
+  modelVoltageSetPointChange->getY0();
+  ASSERT_DOUBLE_EQUALS_DYNAWO(z[0], 0.);
+  ASSERT_DOUBLE_EQUALS_DYNAWO(z[1], 0.);
+  boost::shared_ptr<DataInterface> data;
+  ASSERT_NO_THROW(modelVoltageSetPointChange->initializeFromData(data));
+  ASSERT_NO_THROW(modelVoltageSetPointChange->initializeStaticData());
+  ASSERT_NO_THROW(modelVoltageSetPointChange->checkDataCoherence(0));
+  delete[] zConnected;
+}
+
+TEST(ModelsModelVoltageSetPointChange, ModelVoltageSetPointChangeContinuousAndDiscreteMethods) {
+  boost::shared_ptr<SubModel> modelVoltageSetPointChange = initModelVoltageSetPointChange();
+  std::vector<double> z(modelVoltageSetPointChange->sizeZ(), 0);
+  bool* zConnected = new bool[modelVoltageSetPointChange->sizeZ()];
+  for (size_t i = 0; i < modelVoltageSetPointChange->sizeZ(); ++i)
+    zConnected[i] = true;
+  modelVoltageSetPointChange->setBufferZ(&z[0], zConnected, 0);
+  BitMask* silentZ = new BitMask[modelVoltageSetPointChange->sizeZ()];
+  std::vector<state_g> g(modelVoltageSetPointChange->sizeG(), ROOT_DOWN);
+  modelVoltageSetPointChange->setBufferG(&g[0], 0);
+  modelVoltageSetPointChange->init(0);
+  modelVoltageSetPointChange->getY0();
+  ASSERT_NO_THROW(modelVoltageSetPointChange->setFequations());
+  ASSERT_NO_THROW(modelVoltageSetPointChange->setGequations());
+  std::vector<double> res;
+  ASSERT_NO_THROW(modelVoltageSetPointChange->evalJCalculatedVarI(0, res));
+  std::vector<int> indexes;
+  ASSERT_NO_THROW(modelVoltageSetPointChange->getIndexesOfVariablesUsedForCalculatedVarI(0, indexes));
+  ASSERT_NO_THROW(modelVoltageSetPointChange->evalCalculatedVarI(0));
+  ASSERT_NO_THROW(modelVoltageSetPointChange->evalCalculatedVars());
+  ASSERT_NO_THROW(modelVoltageSetPointChange->evalDynamicFType());
+  ASSERT_NO_THROW(modelVoltageSetPointChange->evalDynamicYType());
+  modelVoltageSetPointChange->collectSilentZ(silentZ);
+  for (size_t i = 0; i < modelVoltageSetPointChange->sizeZ(); ++i) {
+    ASSERT_TRUE(silentZ[i].getFlags(NotUsedInContinuousEquations));
+  }
+
+  modelVoltageSetPointChange->evalF(0, UNDEFINED_EQ);
+  SparseMatrix smj;
+  int size = modelVoltageSetPointChange->sizeF();
+  smj.init(size, size);
+  modelVoltageSetPointChange->evalJt(0, 0, smj, 0);
+  ASSERT_EQ(smj.nbElem(), 0);
+
+  modelVoltageSetPointChange->evalG(2.);
+  modelVoltageSetPointChange->evalZ(2.);
+  modelVoltageSetPointChange->evalF(2., UNDEFINED_EQ);
+  ASSERT_DOUBLE_EQUALS_DYNAWO(z[0], -0.05);
+  ASSERT_DOUBLE_EQUALS_DYNAWO(z[1], -0.05);
+
+  modelVoltageSetPointChange->evalG(6.);
+  modelVoltageSetPointChange->evalZ(6.);
+  modelVoltageSetPointChange->evalF(6., UNDEFINED_EQ);
+  ASSERT_DOUBLE_EQUALS_DYNAWO(z[0], 0.);
+  ASSERT_DOUBLE_EQUALS_DYNAWO(z[1], 0.);
+
+  SparseMatrix smjPrim;
+  smjPrim.init(size, size);
```

Comment:
What is the coverage obtained for the model class?

Reply:
94%